src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -874,6 +877,42 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
         final List<MessageEmbed>       embeds      = map(jsonObject, "embeds",      this::createMessageEmbed);
         final List<MessageReaction>    reactions   = map(jsonObject, "reactions",   (obj) -> createMessageReaction(chan, id, obj));
 
+        if (!jsonObject.isNull("activity"))
+        {
+            JSONObject activityData = jsonObject.getJSONObject("activity");
+            Message.ActivityType activityType = Message.ActivityType.fromId(activityData.getInt("type"));
+            final String partyId = activityData.getString("party_id");
```

Comment:
Use:
optString(..., null) 
Helpers.optInt(..., 0)

We can't be sure what kinda bad data discord might send with new api

Reply:
So always Helpers.optX() or JSONObject.optX()?